onnxruntime/core/providers/cuda/nn/instance_norm.cc

```diff
@@ -100,10 +100,19 @@ Status InstanceNorm<T>::ComputeInternal(OpKernelContext* p_op_kernel_context) co
     CudnnTensor stats_desc;
     ORT_RETURN_IF_ERROR(stats_desc.Set(std::array<int64_t, 4>{1, stats_count, 1, 1}, CudnnTensor::GetDataType<CudaT>()));
 
-    auto mean = GetScratchBuffer<CudaT>(stats_count);
-    auto variance = GetScratchBuffer<CudaT>(stats_count);
-    auto unused_scale = GetScratchBuffer<CudaT>(stats_count);
-    auto unused_bias = GetScratchBuffer<CudaT>(stats_count);
+    size_t stats_buffer_size = stats_count * sizeof(T);
+
+    auto mean = GetScratchBuffer<CudaT>(stats_buffer_size);
+    cudaMemsetAsync(mean.get(), 0, stats_buffer_size, Stream());
```

Comment:
>udaMemsetAsync(mean.get(), 0, stats_buffer_size, Stream() [](http://example.com/codeflow?start=5&length=57)

These are potentially expensive. Please, document the justification.

Reply:
Documented